src/System.Private.CoreLib/src/System/Environment.CoreRT.Unix.cs

```diff
@@ -13,22 +14,71 @@ namespace System
     public static partial class Environment
     {
         internal static int CurrentNativeThreadId => ManagedThreadId.Current;
-        
+
+        private static Dictionary<string, string> _environment;
+
         private static string GetEnvironmentVariableCore(string variable)
         {
             Debug.Assert(variable != null);
-            return Marshal.PtrToStringAnsi(Interop.Sys.GetEnv(variable));
+
+            var environment = LazyInitializer.EnsureInitialized(ref _environment, () => GetSystemEnvironmentVariables());
```

Comment:
It is better to use a help type to cache the value than this, like it is done here: `https://github.com/dotnet/coreclr/blob/master/src/System.Private.CoreLib/src/System/Environment.CoreCLR.cs#L109`

`LazyInitializer` is expensive compared to this - it creates delegate, that delegate is cached in a static variable, ... .

Reply:
Or `Interlocked.CompareExchange`:

```
if (_environment == null)
{
     Interlocked.CompareExchange(_environment, GetSystemEnvironmentVariables(), null);
}
```

This would work well if you switch to initializing on first call to `SetEnvironmentVariable` instead.

Reply:
I was thinking about that, but couldn't find any example of that pattern. Thanks.